### DISCUSSION TITLE
Adaptive streaming with low latency HLS?

### DISCUSSION THREAD

**Discussion by gabridego** [bluenviron/mediamtx#1023] [Questions]
Hello, I am wondering if it is possible to introduce adaptive resolution and bitrate streaming while using low latency HLS, generating several versions of the same live streaming and automatically switching client-side to a lighter or heavier one according to network and CPU condition.
This is natively supported by Apple LLHLS, but I could not find a proper way to achieve it with your application (and neither with FFMPEG when requiring low latency).

Thank you and congrats for the great work!

**Comment by aler9** [comment]
Hello, one of the aim of this server is avoiding re-encoding at all costs in order to save as much CPU as possible. Generating variants requires re-encoding, therefore is not a priority (for the moment). My advice is to tune the original source in terms of bitrate and resolution in order to obtain a versatile (and low latency) stream.
If you absolutely need adaptive resolution for LL-HLS, i think it's implemented in a couple of commercial servers, for instance Wowza and Theo. Keep in mind that it requires computational power, for the reason explained above.

[a comment by an automated account is not shown]
